jwst/associations/lib/rules_level3_base.py

```diff
@@ -294,6 +299,50 @@ def make_member(self, item):
         }
         return member
 
+    def make_fixedslit_bkg(self):
+        """Add a background to a MIR_lrs-fixedslit observation"""
+        for product in self['products']:
+            members = product['members']
+
+            # Split out the science vs. non-science
+            # The non-science exposures will get attached
+            # to every resulting association.
+            science_exps = [
+                member
+                for member in members
+                if member['exptype'] == 'science'
+            ]
+            nonscience_exps = [
+                member
+                for member in members
+                if member['exptype'] != 'science'
+            ]
+            # Create new members for each science exposure in the association,
+            # using the the base name + _x1d as background.
+            results = []
+            new_members = self.current_product['members']
```

Comment:
No longer necessary? You have `members` already for the current product, from line 305. Can you just append back to `members`?

Reply:
code now works directly on the members.